File: atividades/atividade13/ComparaStrings.c
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

void estaContida(char str1[], char str2[]) {
    int tam1 = strlen(str1);
    int tam2 = strlen(str2);

    bool found = false;

    for (int i = 0; i < tam1; i++) {

        if (str1[i] == str2[0]) {
            int check = 1;
            for (int j = 1; j < tam2; j++) {
                if (str1[i + j] == str2[j]) check++;
            }

            if(check == tam2) {
                found = true;
                break;
            }

        }

    }

    if(found) printf("%s contém %s\n", str1, str2);
    else printf("%s não contém %s\n", str1, str2);
}

void inverter(char str[]) {
    int j = strlen(str) - 1;
    int i = 0;
    char temp;

    while(i < j) {
        temp = str[i];
        str[i] = str[j];
        str[j] = temp;
        i++;
        j--;
    }

    printf("%s \n", str);
}

int main(int argc, char *argv[]) {

    int tam1 = strlen(argv[1]);
    int tam2 = strlen(argv[2]);

    char str1[tam1], str2[tam2];

    strcpy(str1, argv[1]);
    strcpy(str2, argv[2]); 

    printf("%s: %d\n", str1, tam1);
    printf("%s: %d\n", str2, tam2);

    estaContida(str1, str2);
    estaContida(str2, str1);

    inverter(str1);
    inverter(str2);

    return 0;
}

File: atividades/atividade13/ComparaStrings.py
import sys

def inverterString(str):
    return ''.join(reversed(str))

def estaContido(str1, str2):

    contem = False
    if len(str1) > len(str2):
        for x in range(len(str1)):
            if str1[x] == str2[0]:
                check = 0
                for y in range(len(str2)):
                    if str1[x + y] == str2[y]:
                        check = check + 1
                if check == len(str2):
                    contem = True
                    break
    
    if contem:
        print(f"{str1} contém {str2}")
    else:
        print(f"{str1} não contém {str2}")

def main():
    print(f"{sys.argv[1]}: {len(sys.argv[1])}")
    print(f"{sys.argv[2]}: {len(sys.argv[2])}")

    estaContido(sys.argv[1], sys.argv[2])
    estaContido(sys.argv[2], sys.argv[1])

    print(inverterString(sys.argv[1]))
    print(inverterString(sys.argv[2]))

if __name__ == '__main__':
    main()